src/main/java/org/jabref/gui/journals/AbbreviateAction.java

```diff
@@ -53,12 +62,24 @@ public void run() {
 
                 return false;
             };
+            tasks.add(callable);
+        }
 
-            boolean result = JabRefExecutorService.INSTANCE.executeAndWait(callable);
-            if (result) {
-                count++;
+        // Execute the callables and wait for the results.
+        List<Future<Boolean>> futures = JabRefExecutorService.INSTANCE.executeAll(tasks);
+
+        // Evaluate the results of the callables.
+        long count = futures.stream().filter(f -> {
+            boolean result;
+            try {
+                result = f.get();
```

Comment:
Instead of the try, catch, finally construct it is actually easier to follow the rule "return as soon as possible" and write something like
````java
try {
   return f.get();
} catch (...) {
   Log;
   return false;
}
````

Reply:
In this case I was looking at this anwser: https://stackoverflow.com/a/17405806 which prefers the way to return at the end of the method. I will change the code as you suggested.